src/PostgREST/DbRequestBuilder.hs

```diff
@@ -50,7 +50,8 @@ readRequest maxRows allRels allProcs apiRequest  =
         (TargetIdent (QualifiedIdentifier s t) ) -> Just (s, t)
         (TargetProc  (QualifiedIdentifier s proc) ) -> Just (s, tName)
           where
-            retType = pdReturnType <$> M.lookup proc allProcs
+            -- TODO: this proc search won't be accurate when the function is overloaded
```

Comment:
Could we check whether all matching procs have Single retvals? If so, choose Singular else SetOf? It's not perfect but it could improve the case if the overloaded procs are somewhat similar.